app/src/main/java/de/lilithwittmann/voicepitchanalyzer/utils/RecordingPaths.java

```diff
@@ -0,0 +1,35 @@
+package de.lilithwittmann.voicepitchanalyzer.utils;
+
+import android.content.Context;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class RecordingPaths {
+    private static String RECORDING_DIR = "recordings";
+    private static String UNFINISHED_RECORDING_DIR = "unfinished_recordings";
+
+    public static Path getUnfinishedRecordingsDirectoryPath(Context context) {
```

Comment:
Just a minor style thing, which you should feel free to take or leave, but it seems like we're often casting the return of the methods in this class to an optional.

Would it simplify things if these methods returned `Option<Path>` rather than mapped it to null?

Reply:
Actually, I mostly rescind my comment - elsewhere we're not mapping an `Optional` of the return of these methods, rather we're just using these methods in the mapping of other `Option`, like here:

```
Optional<Path> recordingPath = Optional.ofNullable(recordingFile)
                                 .map(file -> RecordingPaths.getUnfinishedRecordingPath(getContext(), file));
```

... I guess I just like `Optionals`

Reply:
I agree with you, but I've gotten criticism on other Java projects for using `Optional`s in member variables and return types. I'm not exactly clear why, but it's possibly because the `Optional` itself can also be `null`, so covering all bases would mean essentially null-checking twice.